src/main/scala/org/ergoplatform/ErgoBox.scala

```diff
@@ -38,46 +39,41 @@ import scala.runtime.ScalaRunTime
   * @param additionalRegisters
   */
 class ErgoBox private(
-    override val value: Long,
-    override val proposition: Value[SBoolean.type],
-    override val additionalRegisters: Map[NonMandatoryIdentifier, _ <: EvaluatedValue[_ <: SType]] = Map(),
-    val transactionId: Array[Byte],
-    val index: Short
-) extends ErgoBoxCandidate(value, proposition, additionalRegisters) {
+                       override val value: Long,
+                       override val proposition: Value[SBoolean.type],
+                       override val additionalTokens: Seq[(TokenId, Long)] = Seq(),
+                       override val additionalRegisters: Map[NonMandatoryRegisterId, _ <: EvaluatedValue[_ <: SType]] = Map(),
+                       val transactionId: Array[Byte],
+                       val index: Short
```

Comment:
@kushti why we need index? Each box's position is uniquely defined by `outputCandidates`. At least maybe we don't need to serialize it and save 2 bytes for each box?

Reply:
@ergomorphic a box is a first-class citizen in Ergo. It is living in a state, which is basically an id -> box dictionary. Thus we need for an unique id for every box (so two boxes of the same value and proposition should have different ids), thus we store (unique) transaction id and also an output number in the box.